src/Analyzer/FileVisitor.php

```diff
@@ -41,17 +41,29 @@ public function enterNode(Node $node): void
          * @see FileVisitorTest::test_should_returns_all_dependencies
          */
         if ($node instanceof Node\Expr\StaticCall && method_exists($node->class, 'toString')) {
+            if ('self' === $node->class->toString()) {
+                return;
+            }
             $this->classDescriptionBuilder
                 ->addDependency(new ClassDependency($node->class->toString(), $node->getLine()));
         }
 
         if ($node instanceof Node\Expr\Instanceof_ && method_exists($node->class, 'toString')) {
```

Comment:
Could this condition be expressed implementing a specific method in Node\Expr\Instanceof_?

Reply:
I think not because is there is no method toString it can generate an exception

Reply:
This is the `Node` that the parser has built after parsing the file.
We are using this library to do this:

https://github.com/nikic/PHP-Parser

Reply:
Ok thanks, so you cannot add methods in `Node \ Expr \ StaticCall` and` Node \ Expr \ InstanceOf_` why its are classes of PHP-Parser and so outside of phparkitect, right?

Reply:
they are part of a syntax tree generated by the external library and you can't add anything.